File: app/src/main/java/com/rv882/fastbootjava/sample/data/FastbootDevice.java
package com.rv882.fastbootjava.sample.data;

import androidx.annotation.NonNull;

import com.rv882.fastbootjava.FastbootDeviceContext;
import com.rv882.fastbootjava.FastbootCommand;

public class FastbootDevice {
    @NonNull
    private String deviceId;
	@NonNull
	private String serialNumber;

	private FastbootDevice(@NonNull String deviceId, @NonNull String serialNumber) {
		this.deviceId = deviceId;
		this.serialNumber = serialNumber;
	}

	@NonNull
	public String getDeviceId() {
		return deviceId;
	}

	@NonNull
	public String getSerialNumber() {
		return serialNumber;
	}
	
	@NonNull
	public static FastbootDevice fromDeviceId(@NonNull String deviceId) {
		return new FastbootDevice(deviceId, "");
	}
	
	@NonNull
	public static FastbootDevice fromDeviceContext(@NonNull String deviceId, @NonNull FastbootDeviceContext deviceContext) {
		return new FastbootDevice(deviceId, deviceContext.sendCommand(FastbootCommand.getVar("serialno"), false).getData());
	}
}

File: sample/src/main/java/com/rv882/fastbootjava/sample/data/FastbootDevice.kt

package com.rv882.fastbootjava.sample.data

import com.google.android.fastbootmobile.DeviceId
import com.google.android.fastbootmobile.FastbootCommand
import com.google.android.fastbootmobile.FastbootDeviceContext

data class FastbootDevice(val deviceId: DeviceId, val serialNumber: String, val currentSlot: String) {
    companion object {
        fun fromDeviceId(deviceId: DeviceId) = FastbootDevice(deviceId, "", "")

        fun fromFastbootDeviceContext(deviceId: DeviceId, deviceContext: FastbootDeviceContext): FastbootDevice =
                FastbootDevice(
                        deviceId,
                        deviceContext.sendCommand(FastbootCommand.getVar("serialno")).data,
                        deviceContext.sendCommand(FastbootCommand.getVar("current-slot")).data)
    }
}
